### ISSUE TITLE
AttributeError: 'str' object has no attribute 'module_name'

### ISSUE THREAD

**Issue by mamelara** [LLNL/spack#2578]
```
File "/global/u2/m/mamelara/spack/lib/spack/spack/build_environment.py", line 343, in set_build_environment_variables
    load_module(pkg.spec.architecture.target.module_name)
AttributeError: 'str' object has no attribute 'module_name'
```
Been getting this error when trying to install

**Comment by becker33** [comment]
@mamelara I believe that with the new ArchSpec refactoring, that line should read 

`load_module(pkg.architecture.target.module_name)`

(deleting '.spec'). Can you check that this works for you and create a PR for that change?

**Comment by becker33** [comment]
Closed by #2579
